aea/connections/stub/connection.py

```diff
@@ -145,15 +168,47 @@ def __init__(
 
     def read_envelopes(self) -> None:
         """Receive new envelopes, if any."""
-        line = self.input_file.readline()
-        logger.debug("read line: {!r}".format(line))
-        lines = b""
-        while len(line) > 0:
-            lines += line
-            line = self.input_file.readline()
-        if lines != b"":
-            self._process_line(lines)
+        ok = _lock_file(self.input_file)
```

Comment:
see above. I'd like to see something like:

`with lock_file(self.input_file):`